components/UnifiedDocFeed/utils/getSelectedUrlFilters.ts

```diff
@@ -0,0 +1,80 @@
+import {
+  feedTypeOpts,
+  sortOpts,
+  tagFilters,
+  topLevelFilters,
+} from "../constants/UnifiedDocFilters";
+import { scopeOptions } from "~/config/utils/options";
+
+export type SelectedUrlFilters = {
+  topLevel: string | undefined;
+  type: string | undefined;
+  sort: string | undefined;
+  time: string | undefined;
+  tags: string[];
+};
+
+export const getSelectedUrlFilters = ({
+  query,
+  pathname,
+}): SelectedUrlFilters => {
+  const defaults = {
+    topLevel: topLevelFilters[0].value,
+    type: Object.values(feedTypeOpts)[0].value,
+    sort: sortOpts[0].value,
+    time: scopeOptions[0].value,
+    tags: <string[]>[],
+  };
```

Comment:
keep it outside the component.

Reply:
`defaults` is an object returned as part of the function. Moving it outside of this component will result in polluting the global scope.

It is true that node probably avoids such things however it feels dirty placing a var returning within a function in the global scope (i.e. outside of the function definition).